### ISSUE TITLE
Distributed resnet look like run in serial

### ISSUE THREAD

**Issue by cklsoft** [tensorflow/models#4513]
### System information
- **What is the top-level directory of the model you are using**: official/resnet
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: 
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: linux
- **TensorFlow installed from (source or binary)**: binary
- **TensorFlow version (use command below)**: 1.8.0
- **Bazel version (if compiling from source)**:
- **CUDA/cuDNN version**:
- **GPU model and memory**: 15471MiB, 2 GPUs
- **Exact command to reproduce**: set TF_CONFIG, and run resnet

You can collect some of this information using our environment capture script:

https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh

You can obtain the TensorFlow version with

python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"

### Describe the problem

I ran resnet in multi-gpus in one gpu and 15 gpus, respectively. The result show in worker log is : 

I0613 06:23:03.042901 140540576401216 tf_logging.py:116] loss = 1.5084058, step = 2114
INFO:tensorflow:loss = 1.7659625, step = 3616 (45.301 sec)
I0613 06:23:48.344144 140540576401216 tf_logging.py:116] loss = 1.7659625, step = 3616 (45.301 sec)
INFO:tensorflow:loss = 1.2500832, step = 5124 (45.533 sec)

It cost 45 secs  to run 1500 steps. While in the scenario only use one gpu, the result is:

INFO:tensorflow:loss = 1.4370964, step = 300 (3.163 sec)
I0613 06:35:03.420171 140657753802560 tf_logging.py:116] loss = 1.4370964, step = 300 (3.163 sec)
INFO:tensorflow:global_step/sec: 29.615
I0613 06:35:06.795655 140657753802560 tf_logging.py:116] global_step/sec: 29.615
INFO:tensorflow:loss = 1.3290867, step = 400 (3.377 sec)
I0613 06:35:06.796813 140657753802560 tf_logging.py:116] loss = 1.3290867, step = 400 (3.377 sec)

It seems that, one core (1 worker) cost 3s to run 100 steps, and 15 workers(each worker has one gpu) cost 45s to run 1500steps, 3*15=45! 

### Source code / logs

I modify `resnet_run_loop.resnet_main` to fix the distributed trainning, and add TF_CONFIG in each worker. Since it will hang in distributed mode when using estimator.train, I use estimator.train_and_evaluate. On the other hand, train_and_evaluate only support one iteration, and will cause 'address already in use' in the second iteration, so I use the estimator.train after the first iteration.

`def resnet_main(
    flags_obj, model_function, input_function, dataset_name, shape=None):

   .... # no change

  total_training_cycle = (flags_obj.train_epochs //
                          flags_obj.epochs_between_evals)
  train_spec=tf.estimator.TrainSpec(input_fn=input_fn_train,max_steps=1000)
  eval_spec=tf.estimator.EvalSpec(input_fn=input_fn_eval,steps=1,throttle_secs=1,start_delay_secs=1)
  max_steps=0
  for cycle_index in range(total_training_cycle):
    print("cycle_index....")
    tf.logging.info('Starting a training cycle: %d/%d',
                    cycle_index, total_training_cycle)

    tf.logging.info('Starting to evaluate.')

    max_steps+=train_spec.max_steps
    if cycle_index==0:
      tf.logging.info('Starting to evaluate==.')
      tf.estimator.train_and_evaluate(classifier,train_spec,eval_spec)
    else:
      tf.logging.info('Starting to evaluate+++.')
eval_spec=tf.estimator.EvalSpec(input_fn=input_fn_eval,steps=1,throttle_secs=1,start_delay_secs=1)
      classifier.train(input_fn=train_spec.input_fn,max_steps=max_steps,hooks=list(train_spec.hooks))
    #tf.estimator.train_and_evaluate(classifier,train_spec,eval_spec)

    # print('eval_results: '+str(eval_results))
    # benchmark_logger.log_evaluation_result(eval_results)

    # if model_helpers.past_stop_threshold(
    #     flags_obj.stop_threshold, eval_results['accuracy']):
    #   print('past_stop_threshold break')
    #   break

  if flags_obj.export_dir is not None:
    # Exports a saved model for the given classifier.
    input_receiver_fn = export.build_tensor_serving_input_receiver_fn(
        shape, batch_size=flags_obj.batch_size)
    classifier.export_savedmodel(flags_obj.export_dir, input_receiver_fn)

`

**Comment by MarziEd** [comment]
I have a question from your experiments on 15 GPUs. I need to do a similar experiment on Resnet and I am wondering if the current version of Resnet model and Tf.contrib.distribute module can support multiple GPUs on different machines without any changes to the Resnet code. I ran the code on multiple GPUs on the same machine but not on multiple GPUs on different machines.
